File: 2015/day1NotQuiteLisp/day1.ts
// --- Day 1: Not Quite Lisp ---

import { readFileSync } from "node:fs";

const input: string = "day1Input.txt";

const instructions: string[] = readFileSync(input, {
  encoding: "utf-8",
}).split("");

let floor: number = 0;

instructions.forEach((instruction: string): void => {
  floor += instruction === "(" ? 1 : -1;
});

console.log(`The instructions take Santa to the floor: ${floor}`); //Answer Part 1

// Part 2

floor = 0;

const positionToEnterTheBasement: number =
  instructions.findIndex((instruction: string): boolean => {
    floor += instruction === "(" ? 1 : -1;
    return floor === -1;
  }) + 1;

console.log(
  `The position of the instruction to enter the basement is: ${positionToEnterTheBasement}`
); // Answer Part 2

File: 2015/day1NotQuiteLisp/day1.py
# --- Day 1: Not Quite Lisp ---

input = "day1Input.txt"

with open(input, "r", encoding="utf-8") as file:
    instructions = file.read()

floor = 0

for instrunction in instructions:
    floor += 1 if instrunction == "(" else -1

print(f"The instructions take Santa to the floor: {floor}")  # Answer Part 1

# Part 2

floor = 0

position_to_enter_the_basement = None


for index, instrunction in enumerate(instructions, 1):
    floor += 1 if instrunction == "(" else -1
    if floor == -1:
        position_to_enter_the_basement = index
        break

print(
    f"The position of the instruction to enter the basement is: {position_to_enter_the_basement}"
)  # Answer Part 2
